tigerpath/majors_and_certificates/README.md

```diff
@@ -295,3 +295,158 @@ Here is an example of their usage:
   ]
 }
 ```
+
+### Class Year Differences
+
+The requirements file format supports another primitive called `year_switch`,
+which decides from a list of possibilities depending on the user's class year.
+This is useful when the requirements differ for students in different class
+years.
+For example, the requirements for a certain major might change, but then
+only apply to the incoming class and later, or an exception might be made
+for a specific class year.
+
+A `year_switch` is formatted almost like a `req_list`, in that its value is
+a list of subrequirements.
+Each subrequirement must have a `year_code` field, a string or integer which
+specifies which class year range that the subrequirement applies to
+(refer to the [table below](#year-code-values) for possible values).
+Style dictates that, for readability, the `year_code` field should always
+appear at the top of the subrequirement.
+
+The first requirement in the list whose `year_code` matches the student's
+class year is selected, at which point its explicitly listed fields override
+those of the parent requirement (the one that contains the `year_switch`),
+and the `year_switch` and `year_code` fields are dropped.
+Interpretation of the requirements file then proceeds as usual (with any
+remaining `year_switch`s evaluated recursively).
+
+As an example, this requirement
+
+```yaml
+---
+name: A Transitioning Requirement
+max_counted: 1
+min_needed:
+explanation: Students from the class of 2021 should take any one NST course,
+  while students from the classes of 2022 and later should take one 300-level
+  and one 400-level NST course.
+year_switch:
```

Comment:
Have you considered making this an object instead of an array? I don't know yaml but in json it would be like:

```
"year_switch": {
  "2021": {
    "min_needed": 1
    ...
  },
  ">=2022": {

  },
  ...
}
```

It's a small difference, but it might be better suited because you're only ever taking one of the children (and you also won't have to manually delete the `year_code`).

Reply:
Yeah, this is actually how I did it on first on first try.

The problem with this is that there is no specified order to objects/dictionaries, so the prescription that "the first matching year code" wins is impossible to do in a well-defined way.
In order for that to happen, they have to have a well-defined ordering, which as far as I know can only be done as a list.

Reply:
Ah, good point, forgot about the ordering. An array makes more sense then.